cms/djangoapps/contentstore/management/commands/generate_courses.py

```diff
@@ -0,0 +1,150 @@
+"""
+import json
+import logging
+
+from django.contrib.auth.models import User
+from django.core.management.base import BaseCommand, CommandError
+
+from contentstore.management.commands.utils import user_from_str
+from contentstore.views.course import create_new_course_in_store
+from xmodule.course_module import CourseFields
+from xmodule.fields import Date
+from xmodule.modulestore.exceptions import DuplicateCourseError
+from xmodule.tabs import CourseTabList
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """ Generate a basic course """
+    help = 'Generate courses on studio drom a json list of courses'
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'courses_json',
+        )
+
+    def handle(self, *args, **options):
+        try:
+            courses = json.loads(options["courses_json"])["courses"]
+        except ValueError:
+            raise CommandError("Invalid JSON object")
+        except KeyError:
+            raise CommandError("JSON object is missing courses list")
+
+        for course_settings in courses:
+            # Validate course
+            if not self._course_is_valid(course_settings):
+                logger.warning("Can't create course, proceeding to next course")
+                continue
+
+            # Retrieve settings
+            org = course_settings["organization"]
+            num = course_settings["number"]
+            run = course_settings["run"]
+            user_email = course_settings["user"]
+            try:
+                user = user_from_str(user_email)
+            except User.DoesNotExist:
+                logger.warning(user_email + " user does not exist")
+                logger.warning("Can't create course, proceeding to next course")
+                continue
+            fields = self._process_course_fields(course_settings["fields"])
+
+            # Create the course
+            try:
+                new_course = create_new_course_in_store("split", user, org, num, run, fields)
+                logger.info("Created {}".format(unicode(new_course.id)))
+            except DuplicateCourseError:
+                logger.warning("Course already exists for %s, %s, %s", org, num, run)
+
+    def _process_course_fields(self, fields):
+        """ Returns a validated list of course fields """
+        all_fields = CourseFields.__dict__.keys()
+        non_course_fields = [
+            "__doc__",
+            "__module__",
+            "__weakref__",
+            "__dict__"
+        ]
+        for field in non_course_fields:
+            all_fields.remove(field)
+
+        # Non-primitive course fields
+        date_fields = [
+            "certificate_available_date",
+            "announcement",
+            "enrollment_start",
+            "enrollment_end",
+            "start",
+            "end"
+        ]
+        course_tab_list_fields = [
+            "tabs"
+        ]
+
+        for field in dict(fields):
```

Comment:
* Second case should just be if fields[field] is None, the extra logic must have been something carried over fromwhen I had this setup differently before